compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDmaLoopSubsumption.cpp

```diff
@@ -289,17 +289,36 @@ struct SubsumeLoopIntoDMA
                 insertSourceStrides, updateSourceOffsets))) {
           return failure();
         }
+
+        // Check each dim of the new sizes/strides after insertion to make sure
+        // they are not out of the range.
+        std::optional<SmallVector<int64_t>> maybeNewSourceSizes =
+            getConstantIntValues(newSourceSizes);
+        std::optional<SmallVector<int64_t>> maybeNewSourceStrides =
+            getConstantIntValues(newSourceStrides);
+        assert(maybeNewSourceSizes.has_value() &&
+               "expect constant source size values");
+        assert(maybeNewSourceStrides.has_value() &&
+               "expect constant source stride values");
+        SmallVector<int64_t> newSourceSizesInt = maybeNewSourceSizes.value();
```

Comment:
nit: value() returns a reference, but here you make a copy, consider  `const &`?